libraries/AP_Compass/Compass_learn.h

```diff
@@ -2,6 +2,10 @@
 
 #include <AP_AHRS/AP_AHRS.h>
 
+#define COMPASS_LEARN_NUM_SAMPLES               30
+#define COMPASS_LEARN_BEST_ERROR_THRESHOLD      50
+#define COMPASS_LEARN_WORST_ERROR_THRESHOLD     65
+
```

Comment:
These should be defined in the cpp.  But by putting these up the top of the file you're moving them away from the one place they're used.

If you want to give the constants names, make them `static const COMPASS_LEARN_NUM_SAMPLES = 30;` right above where they're being used.

Reply:
static const uint8_t

Reply:
Thank you for comment. These definitions just want to remove the magic number. I understand that it should be in local scope.